src/rust-crypto/index.ts

```diff
@@ -84,22 +84,29 @@ export async function initRustCrypto(args: {
 
     /** The pickle key for `legacyCryptoStore` */
     legacyPickleKey?: string;
-
     /**
      * A callback which will receive progress updates on migration from `legacyCryptoStore`.
      *
      * Called with (-1, -1) to mark the end of migration.
      */
     legacyMigrationProgressListener?: (progress: number, total: number) => void;
+    /**
+     * Whether to enable tracing. Defaults to true.
+     */
+    tracingEnabled?: boolean;
 }): Promise<RustCrypto> {
-    const { logger } = args;
+    const { logger, tracingEnabled = true } = args;
 
     // initialise the rust matrix-sdk-crypto-wasm, if it hasn't already been done
     logger.debug("Initialising Rust crypto-sdk WASM artifact");
     await RustSdkCryptoJs.initAsync();
 
-    // enable tracing in the rust-sdk
-    new RustSdkCryptoJs.Tracing(RustSdkCryptoJs.LoggerLevel.Debug).turnOn();
+    // Enable tracing in the rust-sdk based on the parameter
+    if (tracingEnabled) {
+        new RustSdkCryptoJs.Tracing(RustSdkCryptoJs.LoggerLevel.Debug).turnOn();
+    } else {
+        // Do not initialize tracing when disabled
+    }
```

Comment:
the problem with this is that incorrectly gives the impression that tracing can be enabled or disabled at the level of an individual instance of `RustCrypto`.

For example: if I call `initRustCrypto({tracingEnabled: true})`, and then later, for a different MatrixClient instance, call `initRustCrypto({tracingEnabled: false})`, tracing will be enabled for the second instance despite `tracngEnabled` being set to false.

What we really need is to do the work in `matrix-sdk-crypto-wasm` to connect the logging from the rust tracing system to `logger`, so that it can be correctly enabled or disabled via the js-sdk's logging API.